### DISCUSSION TITLE
Refine "Continue Reading" link in the blog index page

### DISCUSSION THREAD

**Discussion by wez** [squidfunk/mkdocs-material#6759] [Q&A] [answered]
Hi!

If I use the `<!-- more -->` marker to define the cut point for an excerpt, the index page works as I'd expect; the content before the marker is used as an excerpt and then the "Continue Reading" link takes the visitor to the full content for that post. All good.

However, if I omit the `<!-- more -->` marker then the entire content is used in the index page (which is what I'd expect), AND "Continue Reading" is shown below it.  That gives the visitor the expectation that there is more to read in that case, but when they click through they are disappointed.

What I'd like to be able to do in this second case is to either:
* change the "Continue Reading" text to something like "Permalink to <title>"
* Disable the "Continue Reading" link

is that possible today with a sprinkling of jinja/partials, or is this something that might need code changes in the associated plugin?

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @wez,
yes you would have to modify the `partials/post.html` template, because you want the "Continue Reading" link to be dynamic based on if the `<!-- more -->` is present. Currently it's always being printed, there is no configuration setting to change it:
https://github.com/squidfunk/mkdocs-material/blob/a6286ef3ac3407e8b6c985cf0571fc0e2caa6f5b/material/templates/partials/post.html#L52-L59

Also the `post` in this template is actually an `Excerpt` object. Each `View`/`Post` has an `Excerpt` but there is no "flag" that keeps track if a post has an excerpt with the `post_excerpt_separator` or without it. However, as it stands the `Excerpt` has access to their parent `Post` object, therefore we can create a hacky conditional in the Jinja template:

```jinja
  <div class="md-post__content md-typeset">
    {{ post.content }}
    {% if config.plugins["material/blog"].config.post_excerpt_separator in post.post.content %}
    <nav class="md-post__action">
      <a href="{{ post.url | url }}">
        {{ lang.t("blog.continue") }}
      </a>
    </nav>
    {% endif %}
  </div>
```
`post.post.content` is actually `excerpt.partent_post.content`

> Note that checking if the `post_excerpt_separator` is in the `content` does "scan" the text again and will impact performance relative to the size of the blog/documentation.

> Here is a working example: [blog-basic-continue-reading.zip](https://github.com/squidfunk/mkdocs-material/files/14231335/blog-basic-continue-reading.zip)

**Reply by kamilkrzyskow** [reply]
And here is the second approach with the alternative text inside of the link:
```jinja
  <div class="md-post__content md-typeset">
    {{ post.content }}
    {% if config.plugins["material/blog"].config.post_excerpt_separator in post.post.content %}
        {% set continue_text = lang.t("blog.continue") %}
    {% else %}
        {% set continue_text = "Permalink to " + post.post.title %}
    {% endif %}
    <nav class="md-post__action">
      <a href="{{ post.url | url }}">
        {{ continue_text }}
      </a>
    </nav>
  </div>
```

**Reply by squidfunk** [reply]
I guess we could remove the 'Continue Reading' link in case a post has no separator from the post template. The blog post's title already is a permalink, so we shouldn't add another permanent link at the bottom. You can open a new change request, ideally with a minimal reproduction, as this has now been requested multiple times.

**Reply by kamilkrzyskow** [reply]
Maybe the example with the "Permalink to" was not exactly the right choice, but a "Continue to comments" link/button would be nice.
But this is also risky, as there might be no comments present and the reader will be once again disappointed 😅

**Reply by squidfunk** [reply]
Yeah, so in general we don't add options to the plugin that are solely intended as template switches, as this should be done by customization because of this exact reason – too many possible combinations. However, we need to expose some logic to the template that says "this post is only an excerpt", and then only show the 'Continue reading' link in that case, as that's harder to check in the template. If somebody wants to bring it back or change the logic, customization it is.

**Reply by wez** [reply]
@kamilkrzyskow thanks! your first example works for me!

I would certainly appreciate it if this behavior (not showing the continue link if no excerpt marker is present) is the default behavior, so that I don't have the "worry" of tracking changes to the partial in future releases and keeping things in sync.

I think the zip file linked above is probably all you need for a minimal repro? Do you want me to file the issue for that change request, or are you folks on it?

**Reply by squidfunk** [reply]
Yes, we need a new issue for that, if you want it done. Thank you!
